src/databricks/labs/remorph/reconcile/query_builder.py

```diff
@@ -8,129 +9,93 @@
-
-        if self.table_conf.jdbc_reader_options and self.layer == "source":
-            partition_column = {self.table_conf.jdbc_reader_options.partition_column}
-        else:
-            partition_column = set()
-
-        # Combine all column names
-        all_columns = join_columns | select_columns
-
-        # Remove threshold and drop columns
-        threshold_columns = {thresh.column_name for thresh in self.table_conf.thresholds or []}
-        if self.table_conf.drop_columns is None:
-            drop_columns = set()
-        else:
-            drop_columns = self._get_mapped_columns(self.layer, column_mapping, self.table_conf.drop_columns)
+    def _get_custom_transformation(self, columns, transformation_dict, column_mapping):
+        transformation_rule_mapping = []
+        for column in columns:
+            if column in transformation_dict.keys():
+                transformation = self._get_layer_transform(transformation_dict, column, self.qc.layer)
+            else:
+                transformation = None
 
-        columns = sorted(all_columns - threshold_columns - drop_columns)
-        key_columns = sorted(join_columns | partition_column)
+            column_origin, column_alias = self._get_column_alias(self.qc.layer, column, column_mapping)
 
-        return columns, key_columns
+            transformation_rule_mapping.append(TransformRuleMapping(column_origin, transformation, column_alias))
 
-    def _generate_transformation_rule_mapping(self, columns: list[str], schema: dict) -> list[TransformRuleMapping]:
-        transformations_dict = self.table_conf.list_to_dict(Transformation, "column_name")
-        column_mapping_dict = self.table_conf.list_to_dict(ColumnMapping, "target_name")
+        return transformation_rule_mapping
 
+    def _get_default_transformation(self, columns, column_mapping, schema):
         transformation_rule_mapping = []
         for column in columns:
-            if column_mapping_dict and self.layer == "target":
-                transform_column = (
-                    column_mapping_dict.get(column).source_name if column_mapping_dict.get(column) else column
-                )
-            else:
-                transform_column = column
-
-            if transformations_dict and transform_column in transformations_dict.keys():
-                transformation = self._get_layer_transform(transformations_dict, transform_column, self.layer)
-            else:
-                column_data_type = schema.get(column).data_type
-                transformation = self._get_default_transformation(self.source, column_data_type).format(column)
+            column_origin = column if self.qc.layer == "source" else self._get_column_map(column, column_mapping)
+            column_data_type = schema.get(column_origin).data_type
+            transformation = self._get_default_transformation_expr(self.qc.db_type, column_data_type).format(
```

Comment:
Bug? Is there a format() method on this?

Reply:
it is  a str and it has a placeholder {} to use str.format